compiler-rt/lib/asan/asan_allocator.cpp

```diff
@@ -1540,5 +1547,28 @@ hsa_status_t asan_hsa_amd_vmem_address_free(void* ptr, size_t size,
   }
   return REAL(hsa_amd_vmem_address_free)(ptr, size);
 }
+
+hsa_status_t asan_hsa_amd_pointer_info(const void* ptr,
+                                       hsa_amd_pointer_info_t* info,
+                                       void* (*alloc)(size_t),
+                                       uint32_t* num_agents_accessible,
+                                       hsa_agent_t** accessible) {
+  void* ptr_ = get_allocator().GetBlockBegin(ptr);
+  AsanChunk* m = instance.GetAsanChunkByAddr(reinterpret_cast<uptr>(ptr_));
```

Comment:
Do we not need to guard against null ptr_ here like we do in ipc create?

Reply:
I default initialized status with value `HSA_STATUS_ERROR_NOT_INITIALIZED` in 00898ca25fba0c25f3f1320f7cde6ab07d49434b.

Reply:
> If `ptr_ == nullptr` then `if` condition at line number 1559 will fail.
> 
But is it OK to call GetAsanChunkByAddr with a null ptr_?  Line 1456 avoids doing this.